nondex-test/src/test/java/edu/illinois/nondex/core/QueueTest.java

```diff
@@ -86,7 +87,7 @@ public void testRemoveWhenEmpty() {
     }
 
     @Test(expected = IllegalStateException.class)
-    public void testRemoveWhenNotEmpty() {
+    public void testRemoveProirToNext() {
```

Comment:
there is no next in this test... prior means before and it sort of implies there will be a call to next.
Just say: testRemoveWithoutCallingNextBefore
It's ok for the method name to be long, it's not as if anybody would call this method ever and needs to type its name.